perception/lidar_transfusion/include/lidar_transfusion/preprocess/preprocess_kernel.hpp

```diff
@@ -0,0 +1,72 @@
+// Copyright 2024 TIER IV, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2021 NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved. SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef LIDAR_TRANSFUSION__PREPROCESS__PREPROCESS_KERNEL_HPP_
+#define LIDAR_TRANSFUSION__PREPROCESS__PREPROCESS_KERNEL_HPP_
+
+#include "lidar_transfusion/transfusion_config.hpp"
+#include "lidar_transfusion/utils.hpp"
+
+#include <cuda_runtime_api.h>
+
+namespace lidar_transfusion
+{
+
+class PreprocessCuda
+{
+public:
+  PreprocessCuda(const TransfusionConfig & config, cudaStream_t & stream);
+
+  void generateVoxels(
+    float * points, unsigned int points_size, unsigned int * pillar_num, float * voxel_features,
+    unsigned int * voxel_num, unsigned int * voxel_idxs);
+
+  cudaError_t generateVoxels_random_launch(
+    float * points, unsigned int points_size, unsigned int * mask, float * voxels);
+
+  cudaError_t generateBaseFeatures_launch(
+    unsigned int * mask, float * voxels, unsigned int * pillar_num, float * voxel_features,
+    unsigned int * voxel_num, unsigned int * voxel_idxs);
+
+  cudaError_t generateVoxelsInput_launch(
+    uint8_t * cloud_data, CloudInfo & cloud_info, unsigned int points_agg, unsigned int points_size,
+    float time_lag, float * affine_past2current, float * points);
```

Comment:
In this context, the transformation does not have the meaning of `affine_past2current` so it makes more sense to have a generic name

Reply:
In this context the `affine_past2current` exactly mean affine transformation matrix. Could you check it again?

Reply:
I meant that while in 
https://github.com/amadeuszsz/autoware.universe/blob/63a30098d1c9f6e71e50e96ab0d46acb811a3c05/perception/lidar_transfusion/lib/preprocess/voxel_generator.cpp#L68
The variables has the meaning of a transformation between different time stamps,
in the kernel that nuance is lot and you have kernel that applies a transformation, not necessarily one between two time points. As such, I would recommend a name like `affine_transform`

Reply:
https://github.com/autowarefoundation/autoware.universe/pull/6890/commits/3c0187f3d1bc8aacc640ea313016cff8b2e7aaf1